pkg/platform/runtime/setup/setup.go

```diff
@@ -544,6 +554,36 @@ func (s *Setup) downloadArtifact(unsignedURI string, targetFile string, progress
 	return nil
 }
 
+//downloadArtifact downloads an artifact and returns the local path to that artifact's archive.
+func (s *Setup) downloadArtifact(a artifact.ArtifactDownload, as ArtifactSetuper) (string, error) {
```

Comment:
This function seems to be doing too much. It says "download" in the name of the function but it's also performing setup. Please separate it out such that setup is performed with the result of this function, not inside this function. Keep it unixy.

Reply:
It looks like it's doing more, but it's just downloading. I've changed the `as` parameter to just be the file extension, because that's all that argument was needed for.